Description: use babel7 instead of old babel6
Author: Yadd <yadd@debian.org>
Bug-Debian: https://bugs.debian.org/978051
Forwarded: not-needed
Last-Update: 2021-10-06

--- a/lib/consolidate.js
+++ b/lib/consolidate.js
@@ -1396,9 +1396,9 @@
  *  Rendering function
  */
 function requireReact(module, filename) {
-  var babel = requires.babel || (requires.babel = require('babel-core'));
+  var babel = requires.babel || (requires.babel = require('@babel/core'));
 
-  var compiled = babel.transformFileSync(filename, { presets: [ 'react' ] }).code;
+  var compiled = babel.transformFileSync(filename, { presets: [ '@babel/preset-react' ] }).code;
 
   return module._compile(compiled, filename);
 }
@@ -1409,14 +1409,14 @@
  *  Converting a string into a node module.
  */
 function requireReactString(src, filename) {
-  var babel = requires.babel || (requires.babel = require('babel-core'));
+  var babel = requires.babel || (requires.babel = require('@babel/core'));
 
   if (!filename) filename = '';
   var m = new module.constructor();
   filename = filename || '';
 
   // Compile Using React
-  var compiled = babel.transform(src, { presets: [ 'react' ] }).code;
+  var compiled = babel.transform(src, { presets: [ '@babel/preset-react' ] }).code;
 
   // Compile as a module
   m.paths = module.paths;
--- a/package.json
+++ b/package.json
@@ -28,8 +28,8 @@
   "devDependencies": {
     "arc-templates": "^0.5.3",
     "atpl": ">=0.7.6",
-    "babel-core": "^6.26.3",
-    "babel-preset-react": "^6.24.1",
+    "@babel/core": "*",
+    "@babel/preset-react": "*",
     "bracket-template": "^1.1.5",
     "coffee-script": "^1.12.7",
     "dot": "^1.1.3",
